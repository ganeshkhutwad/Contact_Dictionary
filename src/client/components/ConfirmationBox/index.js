export { default as ConfirmationBox } from './ConfirmationBox';