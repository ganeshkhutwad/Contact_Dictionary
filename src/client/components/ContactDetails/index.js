export { default as ContactDetails } from './ContactDetails';