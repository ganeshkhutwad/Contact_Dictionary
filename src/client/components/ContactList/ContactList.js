/**
@author Ganesh Khutwad
 */
import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import { ContactDetails } from 'components';

const styles = theme => ({
  root: {
    flexGrow: 1,
    margin: 25
  },
  paper: {
    padding: theme.spacing.unit * 2,
    textAlign: 'center',
    color: theme.palette.text.secondary,
  },
});

const getContactDetailLayout = (classes, list, editContact, deleteContact) => {
    return (
        <Grid key={list.id} item xs={12} sm={6}>
            <ContactDetails editContact={editContact} deleteContact={deleteContact} list={list} className={classes.paper} />
        </Grid>
    );
}

const ContactList = ({classes, lists, editContact, deleteContact}) => {
    return (
        <div className={classes.root}>
            <Grid container spacing={24}>
                {
                    lists.map(list => getContactDetailLayout(classes, list, editContact, deleteContact))
                }
            </Grid>
        </div>
    );
}

ContactList.propType = {
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(ContactList);