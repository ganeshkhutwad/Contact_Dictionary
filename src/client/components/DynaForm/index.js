export { default as DynaForm } from './DynaForm';
