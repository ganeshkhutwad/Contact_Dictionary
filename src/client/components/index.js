export { Header } from './Header';
export { ContactList } from './ContactList';
export { ContactDetails } from './ContactDetails';
export { AddButton } from './AddButton';
