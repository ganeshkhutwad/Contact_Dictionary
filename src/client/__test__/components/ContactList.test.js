/**
Test `AddButton` Presentational Component.
@author Ganesh Khutwad
 */
import React from 'react';
import { ContactList } from '../../components/ContactList';

const classes = { };
const lists = [ ];
const editContact = () => { }
const deleteContact = () => { }

describe('`ContactList` Tests', () => {
    it('`ContactList` Snapshot', () => {
        const contactListWrapper = shallow(<ContactList
                                                classes={classes}
                                                lists={lists}
                                                editContact={editContact}
                                                deleteContact={deleteContact}
                                            />);
        expect(contactListWrapper).toMatchSnapshot();
    });
});
