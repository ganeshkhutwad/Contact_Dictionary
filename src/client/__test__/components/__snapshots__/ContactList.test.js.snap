// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`ContactList\` Tests \`ContactList\` Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ContactList)
    classes={Object {}}
    deleteContact={[Function]}
    editContact={[Function]}
    lists={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "paper": "ContactList-paper-2",
        "root": "ContactList-root-1",
      },
      "deleteContact": [Function],
      "editContact": [Function],
      "lists": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "paper": "ContactList-paper-2",
          "root": "ContactList-root-1",
        },
        "deleteContact": [Function],
        "editContact": [Function],
        "lists": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
